uefi-raw/src/time.rs

```diff
@@ -3,6 +3,42 @@
 use bitflags::bitflags;
 use core::fmt::{self, Display, Formatter};
 
+newtype_enum! {
+    /// Error returned by [`Time`] methods if the specified field
+    /// is outside of the valid range.
+    pub enum TimeError: u32 => {
```

Comment:
More than one field could be out of range, so might be better to model this with a struct, with bool fields for each value that could be invalid. (However, also open to the idea that perhaps that's a bit overengineered.)

Reply:
I like this idea.